app/controllers/api/v1/search_results_controller.rb

```diff
@@ -19,8 +14,28 @@ def create
         end
       end
 
+      def index
+        search_result_list = SearchResult.where(user_id: current_user.id).order(:id)
+        pagy, paginated_results = paginated_resources_for(search_result_list)
+
+        search_result_data = SearchResultsSerializer.new(paginated_results).serializable_hash[:data]
+        pagy_meta = meta_from_pagy(pagy)
+
+        render json: { data: search_result_data, meta: pagy_meta }, status: :ok
+      rescue Pagy::OverflowError
```

Comment:
That's odd and unexpected. IS this really required? We do not need to catch every error. Any unhandled error will return a server error which is better than rendering 404 when it is not about not finding a record.